### PR TITLE
[DEVOPS-1149]  Make generated TLS setup configurable

### PR DESCRIPTION
**Pull request by deepfire**
This is a squashed version of PR #3895

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## How to merge

Send the message `bors r+` to merge this PR. For more information, see
[`docs/how-to/bors.md`](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/how-to/bors.md).

### CODE REVIEW

File: scripts/launch/connect-to-cluster/default.nix [hunk 1]
```diff
@@ -82,6 +83,47 @@ let
       -H "Content-Type: application/json; charset=utf-8" \
       "https://${walletListen}/$request_path" "$@"
   '';
+  tlsConfigResultant = {
```
Comment (disassembler) [inline scripts/launch/connect-to-cluster/default.nix]: We should keep the existing names of the CA. Keep what you have in the sample-wallet-config.nix though. I'm only referring to the defaults if you don't use a custom wallet config:
```
Issuer: O=Input Output HK, CN=Cardano SL Self-Signed Root CA
        Validity
            Not Before: Oct 18 02:47:21 2018 GMT
            Not After : Oct 18 02:48:21 2019 GMT
        Subject: O=Input Output HK, CN=Cardano SL Server Node

      Issuer: O=Input Output HK, CN=Cardano SL Self-Signed Root CA
        Validity
            Not Before: Oct 18 02:47:21 2018 GMT
            Not After : Oct 18 02:48:21 2019 GMT
        Subject: O=Input Output HK, CN=Daedalus Wallet

```

Comment (deepfire) [reply scripts/launch/connect-to-cluster/default.nix]: @disassembler, done.

What is the rationale, though?

The certificate chain generation doesn't happen on IOHK's premises, or under IOHK control -- it's only guided by our scripts.  There's probably something to say about certificates relate to legal documents and what this entails.

I do understand the appeal (for the end-user) of having the vague authority of the certs stamped with our insignia, though.

I don't have a strong opinion anyway..

Comment (jmitchell) [reply scripts/launch/connect-to-cluster/default.nix]: > The certificate chain generation doesn't happen on IOHK's premises, or under IOHK control -- it's only guided by our scripts.

That's all true. IMHO the generation of these certs from scripts we maintain is sufficient for keeping the original identifiers. If users ever need to look at the audit the details of the TLS connection these identifiers will help them (and anyone they contact at IOHK) track down relevant context.

> There's probably something to say about how certificates relate to legal documents and what this entails.

If there are concerns about this you can consult with the legal team.

**Pull request merged**